test/testDisasm.cpp

```diff
@@ -174,14 +174,6 @@ TEST(RoundTripSerializationTest, testFunctionSection) {
 
   roundTripFunctionSection(functions);
 
-  std::vector<Instruction> i2 = {{OpCode::INT_PUSH_CONSTANT, 2},
-                                 {OpCode::INT_PUSH_CONSTANT, 2},
-                                 {OpCode::INT_ADD},
-                                 {OpCode::FUNCTION_RETURN},
-                                 END_SECTION};
-  auto f2 = FunctionDef("testName", 5, i2, 1, 2);
-  functions.push_back(f2);
-  EXPECT_THROW(roundTripFunctionSection(functions), DeserializeException);
```

Comment:
What was this test even doing? Purposefully getting the index wrong?

Reply:
Yup! The index of the second function is 5, so `readFunctionSection()` in the deserializer throws an exception due to this:
```c++
 for (uint32_t i = 0; i < functionCount; i++) {
     functions.emplace_back("", -1, std::vector<Instruction>{});
     readFunction(in, functions.back());
     if (functions[i].index != i) {
      throw DeserializeException{"Invalid index"};
     }
   }
```